src/chatwindow/chatwindowdock.cpp

```diff
@@ -1,69 +1,100 @@
 #include "chatwindowdock.h"
-
-#include <QListWidgetItem>
-#include <QKeyEvent>
-
-#include "mainwindow.h"
+#include "qmlchatwidget.h"
 #include "chatwidget.h"
-
-ChatWindowDock::ChatWindowDock(QWidget *pParent): QDockWidget (pParent)
+// ==========================================================================================
+// ==========================================================================================
+//                                                                                CONSTRUCTOR
+ChatWindowDock::ChatWindowDock(QWidget * pParent) :
+    QDockWidget (pParent)
 {
-    setWindowTitle("Chat");
-    mpChatWidget = new ChatWidget;
+    // Build chat widget & connect it to the dock
+    mpChatWidget = new QmlChatWidget;
 
-    connect(mpChatWidget, &ChatWidget::userToConnectSelected,
-            this, &ChatWindowDock::onUserToConnectSelected);
-    connect(mpChatWidget, &ChatWidget::sendMessage,
-            this, &ChatWindowDock::onSendMessage);
+    connect(mpChatWidget, &ChatWidgetInterface::startSharingRequested,
+            this,         &ChatWindowDock::sendRequestOnStartSharing,
+            Qt::UniqueConnection);
 
-    mpChatWidget->setMinimumWidth(pParent->width() / 2);
-    setWidget(mpChatWidget);    
-}
+    connect(mpChatWidget, &ChatWidgetInterface::stopSharingRequested,
+            this,         &ChatWindowDock::sendRequestOnStopSharing,
+            Qt::UniqueConnection);
+
+    connect(mpChatWidget, &ChatWidgetInterface::messageSent,
+            this,         &ChatWindowDock::shareMessageOnSend,
+            Qt::UniqueConnection);
 
+    setWindowTitle("Chat");
```

Comment:
I recommend you to set a variable, for example const char[] windowTitle = "Chat";

Reply:
Done